File: app/src/main/java/net/sunniwell/common/tools/DateTime.java
package net.sunniwell.common.tools;

import java.sql.Timestamp;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import org.codehaus.jackson.util.MinimalPrettyPrinter;
import org.xmlpull.p019v1.XmlPullParser;

public class DateTime {

    /* renamed from: AM */
    public static final int f334AM = 0;
    public static final int AM_PM = 9;
    public static final int DATE = 5;
    public static final int DAY_OF_MONTH = 5;
    public static final int DAY_OF_WEEK = 7;
    public static final int DAY_OF_WEEK_IN_MONTH = 8;
    public static final int DAY_OF_YEAR = 6;
    public static final int DST_OFFSET = 16;
    public static final int ERA = 0;
    public static final int HOUR = 10;
    public static final int HOUR_OF_DAY = 11;
    public static final int MILLISECOND = 14;
    public static final long MILLIS_PER_DAY = 86400000;
    public static final long MILLIS_PER_HOUR = 3600000;
    public static final long MILLIS_PER_MINUTE = 60000;
    public static final int MINUTE = 12;
    public static final int MONTH = 2;
    public static final int SECOND = 13;
    public static final int WEEK_OF_MONTH = 4;
    public static final int WEEK_OF_YEAR = 3;
    public static final int YEAR = 1;
    public static final int ZONE_OFFSET = 15;
    private Calendar calendar;

    public DateTime() {
        this.calendar = null;
        this.calendar = Calendar.getInstance();
    }

    public DateTime(long lMillis) {
        this();
        this.calendar.setTimeInMillis(lMillis);
    }

    public DateTime(int year, int month, int day) {
        this();
        this.calendar.set(year, month, day);
    }

    public DateTime(int year, int month, int day, int hourOfDay, int minute) {
        this();
        this.calendar.set(year, month, day, hourOfDay, minute);
    }

    public DateTime(int year, int month, int day, int hourOfDay, int minute, int second) {
        this();
        this.calendar.set(year, month, day, hourOfDay, minute, second);
    }

    public DateTime(String strDateTime) {
        this.calendar = null;
        if (strDateTime == null || strDateTime.trim().length() == 0) {
            this.calendar = null;
            return;
        }
        String strDateTime2 = checkStr(strDateTime);
        this.calendar = Calendar.getInstance();
        this.calendar.set(getInt(strDateTime2, 0, 4), getInt(strDateTime2, 4, 2) - 1, getInt(strDateTime2, 6, 2), getInt(strDateTime2, 8, 2), getInt(strDateTime2, 10, 2), getInt(strDateTime2, 12, 2));
    }

    public int getYear() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(1);
    }

    public int getMonth() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(2) + 1;
    }

    public int getDay() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(5);
    }

    public int getWeek() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(7);
    }

    public int getHour() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(11);
    }

    public int getMinute() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(12);
    }

    public int getSecond() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(13);
    }

    public int getMillisecond() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(14);
    }

    public long getTime() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.getTime().getTime();
    }

    public int getTimeZone() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.get(15) / 3600000;
    }

    public String formatDefault() {
        return format("yyyy-MM-dd hh:mm:ss.SSS");
    }

    public String format(String format) {
        if (format == null) {
            return XmlPullParser.NO_NAMESPACE;
        }
        return new SimpleDateFormat(format).format(new Date(getTime()));
    }

    public Timestamp getTimestamp() {
        return new Timestamp(this.calendar.getTimeInMillis());
    }

    public Calendar getCalendar() {
        return this.calendar;
    }

    public long getDateTime() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.getTimeInMillis();
    }

    public long getDateTimeInMillis() {
        if (this.calendar == null) {
            return -1;
        }
        return this.calendar.getTimeInMillis();
    }

    public String getFormattedDateTime() {
        return getFormattedDateTime("yyyy年MM月dd日HH时mm分ss秒");
    }

    public String getFormatDateTime() {
        return getFormattedDateTime("yyyy-MM-dd HH:mm:ss");
    }

    public String getFormattedDateTime(String strPattern) {
        if (this.calendar == null) {
            return XmlPullParser.NO_NAMESPACE;
        }
        if (strPattern == null || strPattern.trim().length() == 0) {
            return null;
        }
        return new SimpleDateFormat(strPattern).format(this.calendar.getTime());
    }

    public DateTime getFirstDayOfMouth() {
        DateTime dt = new DateTime(getDateTimeInMillis());
        dt.set(5, this.calendar.getActualMinimum(5));
        return dt;
    }

    public DateTime getLastDayOfMouth() {
        DateTime dt = new DateTime(getDateTimeInMillis());
        dt.set(5, this.calendar.getActualMaximum(5));
        return dt;
    }

    public DateTime getFirstDayOfWeek() {
        int iDayOfWeek = this.calendar.get(7);
        DateTime dt = new DateTime(getDateTimeInMillis());
        dt.addDay(1 - iDayOfWeek);
        return dt;
    }

    public DateTime getLastDayOfWeek() {
        DateTime dt = getFirstDayOfWeek();
        dt.addDay(6);
        return dt;
    }

    public void setDateTime(long lMillis) {
        if (this.calendar != null) {
            this.calendar.setTimeInMillis(lMillis);
        }
    }

    public void setDateTime(String strDateTime) {
        if (this.calendar != null) {
            String strDateTime2 = checkStr(strDateTime);
            this.calendar.set(getInt(strDateTime2, 0, 4), getInt(strDateTime2, 4, 2) - 1, getInt(strDateTime2, 6, 2), getInt(strDateTime2, 8, 2), getInt(strDateTime2, 10, 2), getInt(strDateTime2, 12, 2));
        }
    }

    public String getDateTimeString() {
        if (this.calendar == null) {
            return XmlPullParser.NO_NAMESPACE;
        }
        StringBuffer buffer = new StringBuffer();
        buffer.append(this.calendar.get(1));
        buffer.append(padChar(this.calendar.get(2) + 1, 2));
        buffer.append(padChar(this.calendar.get(5), 2));
        buffer.append(padChar(this.calendar.get(11), 2));
        buffer.append(padChar(this.calendar.get(12), 2));
        buffer.append(padChar(this.calendar.get(13), 2));
        return buffer.toString();
    }

    public int get(int field) {
        return this.calendar.get(field);
    }

    public void set(int field, int value) {
        this.calendar.set(field, value);
    }

    public int getActualMaximum(int field) {
        return this.calendar.getActualMaximum(field);
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    public DateTime addYear(int iYear) {
        if (this.calendar == null) {
            return null;
        }
        this.calendar.add(1, iYear);
        return this;
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    public DateTime addMonth(int iMonth) {
        if (this.calendar == null) {
            return null;
        }
        this.calendar.add(2, iMonth);
        return this;
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    public DateTime addDay(int iDay) {
        if (this.calendar == null) {
            return null;
        }
        this.calendar.add(5, iDay);
        return this;
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    public DateTime addHour(int iHour) {
        if (this.calendar == null) {
            return null;
        }
        this.calendar.add(11, iHour);
        return this;
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    public DateTime addMinute(int iMinute) {
        if (this.calendar == null) {
            return null;
        }
        this.calendar.add(12, iMinute);
        return this;
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    public DateTime addSecond(int iSecond) {
        if (this.calendar == null) {
            return null;
        }
        this.calendar.add(13, iSecond);
        return this;
    }

    public static double diffDays(String strStartDate, String strEndDate) {
        return ((double) (new DateTime(strEndDate).getDateTime() - new DateTime(strStartDate).getDateTime())) / 8.64E7d;
    }

    public static double diffDays(DateTime startDate, DateTime endDate) {
        return ((double) (endDate.getDateTime() - startDate.getDateTime())) / 8.64E7d;
    }

    public static double diffHours(String strStartDate, String strEndDate) {
        return diffHours(new DateTime(strStartDate), new DateTime(strEndDate));
    }

    public static double diffHours(DateTime startDate, DateTime endDate) {
        return (((double) (endDate.getDateTime() - startDate.getDateTime())) * 1.0d) / 3600000.0d;
    }

    public static double diffMinutes(String strStartDate, String strEndDate) {
        return diffMinutes(new DateTime(strStartDate), new DateTime(strEndDate));
    }

    public static double diffMinutes(DateTime startDate, DateTime endDate) {
        return (((double) (endDate.getDateTime() - startDate.getDateTime())) * 1.0d) / 60000.0d;
    }

    public String toString() {
        return getDateTimeString();
    }

    private int getInt(String str, int iStart, int iWidth) {
        int iLen = str.length();
        if (iLen < 1 || iLen < iStart || iLen < iStart + iWidth) {
            return 0;
        }
        return new Integer(str.substring(iStart, iStart + iWidth)).intValue();
    }

    private String checkStr(String strDateTime) {
        return padChar(strDateTime.replaceAll("-", XmlPullParser.NO_NAMESPACE).replaceAll(MinimalPrettyPrinter.DEFAULT_ROOT_VALUE_SEPARATOR, XmlPullParser.NO_NAMESPACE).replaceAll(":", XmlPullParser.NO_NAMESPACE).replaceAll("/", XmlPullParser.NO_NAMESPACE).trim(), 14, false);
    }

    private String padChar(int iInfo, int iWidth) {
        String strInfo = String.valueOf(iInfo);
        StringBuffer sbTmp = new StringBuffer(strInfo);
        if (strInfo == null || strInfo.length() > iWidth) {
            return strInfo;
        }
        for (int i = 0; i < iWidth - strInfo.length(); i++) {
            sbTmp.insert(0, '0');
        }
        return sbTmp.toString();
    }

    private String padChar(String strInfo, int iWidth, boolean bAtBegin) {
        StringBuffer sbTmp = new StringBuffer(strInfo);
        if (strInfo == null || strInfo.length() > iWidth) {
            return strInfo;
        }
        for (int i = 0; i < iWidth - strInfo.length(); i++) {
            sbTmp.insert(bAtBegin ? 0 : sbTmp.length(), '0');
        }
        return sbTmp.toString();
    }
}

File: app/src/main/java/com/eliving/tv/common/DateTime.kt
package com.eliving.tv.common

import org.codehaus.jackson.util.MinimalPrettyPrinter
import org.xmlpull.v1.XmlPullParser
import java.sql.Timestamp
import java.text.SimpleDateFormat
import java.util.*

class DateTime {
    var calendar: Calendar?
        private set

    constructor() {
        calendar = null
        calendar = Calendar.getInstance()
    }

    constructor(lMillis: Long) : this() {
        calendar!!.timeInMillis = lMillis
    }

    constructor(year: Int, month: Int, day: Int) : this() {
        calendar!![year, month] = day
    }

    constructor(year: Int, month: Int, day: Int, hourOfDay: Int, minute: Int) : this() {
        calendar!![year, month, day, hourOfDay] = minute
    }

    constructor(
        year: Int,
        month: Int,
        day: Int,
        hourOfDay: Int,
        minute: Int,
        second: Int
    ) : this() {
        calendar!![year, month, day, hourOfDay, minute] = second
    }

    constructor(strDateTime: String?) {
        calendar = null
        if (strDateTime == null || strDateTime.trim { it <= ' ' }.length == 0) {
            calendar = null
            return
        }
        val strDateTime2 = checkStr(strDateTime)
        calendar = Calendar.getInstance()
        calendar?.set(
            getInt(strDateTime2, 0, 4),
            getInt(strDateTime2, 4, 2) - 1,
            getInt(strDateTime2, 6, 2),
            getInt(strDateTime2, 8, 2),
            getInt(strDateTime2, 10, 2),
            getInt(strDateTime2, 12, 2)
        )
    }

    val year: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.YEAR]

    val month: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.MONTH] + 1

    val day: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.DAY_OF_MONTH]

    val week: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.DAY_OF_WEEK]

    val hour: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.HOUR_OF_DAY]

    val minute: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.MINUTE]

    val second: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.SECOND]

    val millisecond: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.MILLISECOND]

    val time: Long
        get() = if (calendar == null) {
            -1
        } else calendar!!.time.time

    val timeZone: Int
        get() = if (calendar == null) {
            -1
        } else calendar!![Calendar.ZONE_OFFSET] / 3600000

    fun formatDefault(): String {
        return format("yyyy-MM-dd hh:mm:ss.SSS")
    }

    fun format(format: String?): String {
        return if (format == null) {
            XmlPullParser.NO_NAMESPACE
        } else SimpleDateFormat(format).format(Date(time))
    }

    val timestamp: Timestamp
        get() = Timestamp(calendar!!.timeInMillis)

    var dateTime: Long
        get() = if (calendar == null) {
            -1
        } else calendar!!.timeInMillis
        set(lMillis) {
            if (calendar != null) {
                calendar!!.timeInMillis = lMillis
            }
        }

    val dateTimeInMillis: Long
        get() = if (calendar == null) {
            -1
        } else calendar!!.timeInMillis

    val formattedDateTime: String?
        get() = getFormattedDateTime("yyyy年MM月dd日HH时mm分ss秒")

    val formatDateTime: String?
        get() = getFormattedDateTime("yyyy-MM-dd HH:mm:ss")

    fun getFormattedDateTime(strPattern: String?): String? {
        if (calendar == null) {
            return XmlPullParser.NO_NAMESPACE
        }
        return if (strPattern == null || strPattern.trim { it <= ' ' }.length == 0) {
            null
        } else SimpleDateFormat(strPattern).format(calendar!!.time)
    }

    val firstDayOfMouth: DateTime
        get() {
            val dt =
                DateTime(dateTimeInMillis)
            dt[5] = calendar!!.getActualMinimum(Calendar.DAY_OF_MONTH)
            return dt
        }

    val lastDayOfMouth: DateTime
        get() {
            val dt =
                DateTime(dateTimeInMillis)
            dt[5] = calendar!!.getActualMaximum(Calendar.DAY_OF_MONTH)
            return dt
        }

    val firstDayOfWeek: DateTime
        get() {
            val iDayOfWeek = calendar!![Calendar.DAY_OF_WEEK]
            val dt =
                DateTime(dateTimeInMillis)
            dt.addDay(1 - iDayOfWeek)
            return dt
        }

    val lastDayOfWeek: DateTime
        get() {
            val dt = firstDayOfWeek
            dt.addDay(6)
            return dt
        }

    fun setDateTime(strDateTime: String) {
        if (calendar != null) {
            val strDateTime2 = checkStr(strDateTime)
            calendar!![getInt(strDateTime2, 0, 4), getInt(strDateTime2, 4, 2) - 1, getInt(
                strDateTime2,
                6,
                2
            ), getInt(
                strDateTime2,
                8,
                2
            ), getInt(strDateTime2, 10, 2)] = getInt(strDateTime2, 12, 2)
        }
    }

    val dateTimeString: String
        get() {
            if (calendar == null) {
                return XmlPullParser.NO_NAMESPACE
            }
            val buffer = StringBuffer()
            buffer.append(calendar!![Calendar.YEAR])
            buffer.append(padChar(calendar!![Calendar.MONTH] + 1, 2))
            buffer.append(padChar(calendar!![Calendar.DAY_OF_MONTH], 2))
            buffer.append(padChar(calendar!![Calendar.HOUR_OF_DAY], 2))
            buffer.append(padChar(calendar!![Calendar.MINUTE], 2))
            buffer.append(padChar(calendar!![Calendar.SECOND], 2))
            return buffer.toString()
        }

    operator fun get(field: Int): Int {
        return calendar!![field]
    }

    operator fun set(field: Int, value: Int) {
        calendar!![field] = value
    }

    fun getActualMaximum(field: Int): Int {
        return calendar!!.getActualMaximum(field)
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    fun addYear(iYear: Int): DateTime? {
        if (calendar == null) {
            return null
        }
        calendar!!.add(Calendar.YEAR, iYear)
        return this
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    fun addMonth(iMonth: Int): DateTime? {
        if (calendar == null) {
            return null
        }
        calendar!!.add(Calendar.MONTH, iMonth)
        return this
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    fun addDay(iDay: Int): DateTime? {
        if (calendar == null) {
            return null
        }
        calendar!!.add(Calendar.DAY_OF_MONTH, iDay)
        return this
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    fun addHour(iHour: Int): DateTime? {
        if (calendar == null) {
            return null
        }
        calendar!!.add(Calendar.HOUR_OF_DAY, iHour)
        return this
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    fun addMinute(iMinute: Int): DateTime? {
        if (calendar == null) {
            return null
        }
        calendar!!.add(Calendar.MINUTE, iMinute)
        return this
    }

    /* Debug info: failed to restart local var, previous not found, register: 2 */
    fun addSecond(iSecond: Int): DateTime? {
        if (calendar == null) {
            return null
        }
        calendar!!.add(Calendar.SECOND, iSecond)
        return this
    }

    override fun toString(): String {
        return dateTimeString
    }

    private fun getInt(str: String?, iStart: Int, iWidth: Int): Int {
        val iLen = str!!.length
        return if (iLen < 1 || iLen < iStart || iLen < iStart + iWidth) {
            0
        } else str.substring(iStart, iStart + iWidth).toInt()
    }

    private fun checkStr(strDateTime: String): String? {
        return padChar(
            strDateTime.replace("-".toRegex(), XmlPullParser.NO_NAMESPACE).replace(
                MinimalPrettyPrinter.DEFAULT_ROOT_VALUE_SEPARATOR.toRegex(),
                XmlPullParser.NO_NAMESPACE
            ).replace(":".toRegex(), XmlPullParser.NO_NAMESPACE)
                .replace("/".toRegex(), XmlPullParser.NO_NAMESPACE)
                .trim { it <= ' ' },
            14,
            false
        )
    }

    private fun padChar(iInfo: Int, iWidth: Int): String? {
        val strInfo = iInfo.toString()
        val sbTmp = StringBuffer(strInfo)
        if (strInfo == null || strInfo.length > iWidth) {
            return strInfo
        }
        for (i in 0 until iWidth - strInfo.length) {
            sbTmp.insert(0, '0')
        }
        return sbTmp.toString()
    }

    private fun padChar(
        strInfo: String?,
        iWidth: Int,
        bAtBegin: Boolean
    ): String? {
        val sbTmp = StringBuffer(strInfo!!)
        if (strInfo == null || strInfo.length > iWidth) {
            return strInfo
        }
        for (i in 0 until iWidth - strInfo.length) {
            sbTmp.insert(if (bAtBegin) 0 else sbTmp.length, '0')
        }
        return sbTmp.toString()
    }

    companion object {
        /* renamed from: AM */
        const val f334AM = 0
        const val AM_PM = 9
        const val DATE = 5
        const val DAY_OF_MONTH = 5
        const val DAY_OF_WEEK = 7
        const val DAY_OF_WEEK_IN_MONTH = 8
        const val DAY_OF_YEAR = 6
        const val DST_OFFSET = 16
        const val ERA = 0
        const val HOUR = 10
        const val HOUR_OF_DAY = 11
        const val MILLISECOND = 14
        const val MILLIS_PER_DAY: Long = 86400000
        const val MILLIS_PER_HOUR: Long = 3600000
        const val MILLIS_PER_MINUTE: Long = 60000
        const val MINUTE = 12
        const val MONTH = 2
        const val SECOND = 13
        const val WEEK_OF_MONTH = 4
        const val WEEK_OF_YEAR = 3
        const val YEAR = 1
        const val ZONE_OFFSET = 15
        fun diffDays(strStartDate: String?, strEndDate: String?): Double {
            return (DateTime(strEndDate)
                .dateTime - DateTime(strStartDate)
                .dateTime).toDouble() / 8.64E7
        }

        fun diffDays(
            startDate: DateTime,
            endDate: DateTime
        ): Double {
            return (endDate.dateTime - startDate.dateTime).toDouble() / 8.64E7
        }

        fun diffHours(strStartDate: String?, strEndDate: String?): Double {
            return diffHours(
                DateTime(
                    strStartDate
                ), DateTime(strEndDate)
            )
        }

        fun diffHours(
            startDate: DateTime,
            endDate: DateTime
        ): Double {
            return (endDate.dateTime - startDate.dateTime).toDouble() * 1.0 / 3600000.0
        }

        fun diffMinutes(strStartDate: String?, strEndDate: String?): Double {
            return diffMinutes(
                DateTime(
                    strStartDate
                ), DateTime(strEndDate)
            )
        }

        fun diffMinutes(
            startDate: DateTime,
            endDate: DateTime
        ): Double {
            return (endDate.dateTime - startDate.dateTime).toDouble() * 1.0 / 60000.0
        }
    }
}